libs/fractals/CurveFractal.cpp

```diff
@@ -0,0 +1,46 @@
+
+#include "CurveFractal.h"
+
+namespace soo {
+
+CurveFractal::CurveFractal() : edge_length_(10.f), rotation_angle_(90.f) {}
+
+CurveFractal::CurveFractal(const float rotation_angle)
+    : edge_length_(10.f), rotation_angle_(rotation_angle) {}
+
+CurveFractal::CurveFractal(const float edge_length, const float rotation_angle)
+    : edge_length_(edge_length), rotation_angle_(rotation_angle) {}
+
+void CurveFractal::GenerateRecursive(unsigned int iteration) {
+  if (iteration > 0) {
+    GenerateNextIteration();
+    GenerateRecursive(--iteration);
+  }
+}
+
+void CurveFractal::GenerateNextIteration() {
+  line_.getVertices().empty() ? Init() : Iterate();
+}
+
+void CurveFractal::draw(const float margin) const {
+  ofPushMatrix();
+  {
+    const ofRectangle& bbox = line_.getBoundingBox();
+
+    // Define scaling parameters
+    float scaleFactor = bbox.width > bbox.height
```

Comment:
shall be `const` and naming `kScaleFactor`

Reply:
it doesn't need to be const, this I take more as a style option. However, the variable naming shall be changed accordingly: either `kScaleFactor` or `scale_factor` but not mixed case.